mne/io/edf/edf.py

```diff
@@ -511,16 +481,529 @@ def _get_edf_info(fname, stim_channel, annot, annotmap, eog, misc, exclude,
+                             for filt in prefiltering])
+        lowpass = np.ravel([re.findall('LP:\s+(\w+)', filt)
+                            for filt in prefiltering])
+
+        # number of samples per record
+        n_samps = np.array([int(fid.read(8).decode()) for ch
+                            in channels])
+
+        # Populate edf_info
+        edf_info['ch_names'] = ch_names
+        edf_info['data_offset'] = header_nbytes
+        edf_info['date'] = date
+        edf_info['digital_max'] = digital_max
+        edf_info['digital_min'] = digital_min
+        edf_info['exclude'] = exclude
+        edf_info['highpass'] = highpass
+        edf_info['include'] = include
+        edf_info['lowpass'] = lowpass
+        edf_info['n_records'] = n_records
+        edf_info['n_samps'] = n_samps
+        edf_info['nchan'] = nchan
+        edf_info['physical_max'] = physical_max
+        edf_info['physical_min'] = physical_min
+        edf_info['record_length'] = record_length
+        edf_info['subtype'] = subtype
+
+        fid.read(32 * nchan).decode()  # reserved
+        assert fid.tell() == header_nbytes
+
+        fid.seek(0, 2)
+        n_bytes = fid.tell()
+        n_data_bytes = n_bytes - header_nbytes
+        total_samps = (n_data_bytes // 3 if subtype == '24BIT'
+                       else n_data_bytes // 2)
+        read_records = total_samps // np.sum(n_samps)
+        if n_records != read_records:
+            warn('Number of records from the header does not match the file '
+                 'size (perhaps the recording was not stopped before exiting).'
+                 ' Inferring from the file size.')
+            edf_info['n_records'] = n_records = read_records
+
+        if subtype in ('24BIT', 'bdf'):
+            edf_info['data_size'] = 3  # 24-bit (3 byte) integers
+        else:
+            edf_info['data_size'] = 2  # 16-bit (2 byte) integers
+
+    return edf_info
+
+
+def _read_gdf_header(fname, stim_channel, exclude):
+    """Read GDF 1.x and GDF 2.x header info."""
+    edf_info = dict()
+    # edf_info['annot'] = annot
+    # edf_info['annotmap'] = annotmap
+    edf_info['events'] = []
+
+    with open(fname, 'rb') as fid:
+        assert(fid.tell() == 0)
+
+        version = fid.read(8).decode()
+
+        gdftyp_np = (None, np.int8, np.uint8, np.int16, np.uint16, np.int32,
+                     np.uint32, np.int64, np.uint64, None, None, None, None,
+                     None, None, None, np.float32, np.float64)
+        gdftyp_byte = [np.dtype(x).itemsize if x is not None else 0
+                       for x in gdftyp_np]
+        assert sum(gdftyp_byte) == 42
+
+        edf_info['type'] = edf_info['subtype'] = version[:3]
+        edf_info['number'] = float(version[4:])
+
+        # GDF 1.x
+        # ----------------------------------------------------------------------
+        if edf_info['number'] < 1.9:
+
+            fid.read(80)  # subject id
+            fid.read(80)  # recording id
```

Comment:
we have `info['subject']` we could put these variables in, have you looked at it?